NXP/MCIMX8M_EVK_4GB/PlatformBuild.py

```diff
@@ -5,21 +5,238 @@
 PROJECT_SCOPE = ('imxfamily', 'imx8')
 
-MODULE_PKGS = ('MU_BASECORE','Silicon/ARM/NXP', 'Common/MU','Common/MU_TIANO', 'Silicon/ARM/MU_TIANO')
+MODULE_PKGS = ('MU_BASECORE','Silicon/ARM/NXP', 'Common/MU','Common/MU_TIANO', 'Common/MU_OEM_SAMPLE','Silicon/ARM/MU_TIANO')
 MODULE_PKG_PATHS = os.pathsep.join(os.path.join(WORKSPACE_PATH, pkg_name) for pkg_name in MODULE_PKGS)
+
+
 #
-#==========================================================================
+# Supported Profiles and Settings based on the profile.  This allows easier grouping
+# of settings.  Add var name as the key and value as value.
+#
+gProfile = {
+    "SECURE": {"BLD_*_CONFIG_AUTH_VAR" : "TRUE", "BLD_*_CONFIG_SECURE_BOOT" : "TRUE",  "BLD_*_CONFIG_MEASURED_BOOT" : "TRUE"},
+    "DEV": {"BLD_*_CONFIG_AUTH_VAR" : "FALSE", "BLD_*_CONFIG_SECURE_BOOT" : "FALSE",  "BLD_*_CONFIG_MEASURED_BOOT" : "FALSE", "BLD_*_CONFIG_OPTEE_PROFILE" : "FALSE"},
+    "FRONTPAGE": {"BLD_*_CONFIG_AUTH_VAR" : "FALSE", "BLD_*_CONFIG_SECURE_BOOT" : "FALSE",  "BLD_*_CONFIG_OPTEE_PROFILE" : "FALSE",  "BLD_*_CONFIG_USB" :"TRUE","BLD_*_CONFIG_MEASURED_BOOT" : "FALSE", "BLD_*_CONFIG_FRONTPAGE" : "TRUE"},
+    }
+
+#--------------------------------------------------------------------------------------------------------
+# Subclass the UEFI builder and add platform specific functionality.
 #
+class PlatformBuilder(UefiBuilder):
+
+    def __init__(self, WorkSpace, PackagesPath, PInManager, PInHelper, args):
+        super(PlatformBuilder, self).__init__(WorkSpace, PackagesPath, PInManager, PInHelper, args)
+
+    def SetPlatformEnv(self):
+        logging.debug("PlatformBuilder SetPlatformEnv")
+
+        # CHECK FOR PROFILE USAGE. SET TO DEV IF NONE IS SPECIFIED.
+        profile = self.env.GetValue("PROFILE")
+        if((profile == None) or (profile.upper() == "DEFAULT")):
+            profile = "DEV"
+        profile = profile.upper()
+        logging.critical("BUILD PROFILE SET: %s" % profile)
+        p = gProfile[profile]
+        if(p != None):
+            for k,v in p.items():
+                logging.debug("Setting %s = %s by using profile %s" % (k, v, self.env.GetValue("PROFILE")))
+                self.env.SetValue(k, v, "PROFILE VALUE")
+        else:
+            #catch case user requested a profile that doesn't exist
+            logging.critical("Profile: %s doesn't exist." % profile)
+            return -1
+        #END OF PROFILE SUPPORT
+
+        self.env.SetValue("BLD_*_CONFIG_HEADLESS", "FALSE", "Display surface will be drawn by default")
+
+        # # Useful if working with Lauterbach JTAG. Will dump to console the complete
+        # # command to copy and past into the Lauterbach command line to load symbols
+        # # for each module.
+        # # Use in conjunction with Informational debug messages:
+        # # DEBUG_INFO      0x00000040
+        self.env.SetValue("BLD_*_CONFIG_DUMP_SYMBOL_INFO", "TRUE", "On by default")
```

Comment:
Can this be made shared code?

Reply:
We have seen customers create a shared module and we have seen customers create a super class, so whatever works for you!